projects/js-packages/publicize-components/src/hooks/use-refresh-connections/index.js

```diff
@@ -0,0 +1,30 @@
+import { useDebounce } from '@wordpress/compose';
+import { useRef } from '@wordpress/element';
+import { usePageVisibility } from 'react-page-visibility';
+import useSelectSocialMediaConnections from '../use-social-media-connections';
+
+/**
+ * Hook that provides a function to refresh the connections.
+ *
+ * @returns { object } The refreshAutoConversionSettings function.
+ */
+export default function useRefreshConnections() {
+	const shouldAutoRefresh = useRef( false );
+	const pageHasFocus = usePageVisibility();
+	const { refresh: refreshConnections } = useSelectSocialMediaConnections();
+	const debouncedRefresh = useDebounce( refreshConnections, 2000 );
+
+	const refresh = () => {
+		if ( ! pageHasFocus ) {
+			shouldAutoRefresh.current = true;
+			debouncedRefresh.cancel();
+		}
+
+		if ( pageHasFocus && shouldAutoRefresh.current ) {
+			debouncedRefresh();
+			shouldAutoRefresh.current = false;
+		}
+	};
+
+	return { refresh };
```

Comment:
May be we can simply return the callback wrapped in `useCallback` instead of an object?

Reply:
I've made it so the function is returned directly. I don't think `useCallback` would help here as the function isn't used as a prop and so wouldn't trigger a render.